IPython/terminal/ptshell.py

```diff
@@ -35,6 +35,7 @@
 from .pt_inputhooks import get_inputhook_func
 from .interactiveshell import get_default_editor, TerminalMagics
 
+_use_simple_prompt = 'IPY_TEST_SIMPLE_PROMPT' in os.environ or not sys.stdin.isatty()
```

Comment:
I think the command line flag should take precedence of the env variable,  especially since it has the `--no-simple-prompt` alternative. IIRC that's how most of our commands works. 

I'm fine changing if you still think it should be.